### PR TITLE
Add check for huge pages

### PR DESCRIPTION
**Pull request by meikomeik**

### CODE REVIEW

File: src/check_huge_pages.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import sys
+
+from argparse import ArgumentParser
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--value',
+                        help='The amout of huge pages to expect')
```
Comment (lamaral) [inline src/check_huge_pages.py:34]: Please make the argument required, as the script can't run without it.

File: src/check_huge_pages.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if not meminfo:
+        print('Could not parse /proc/meminfo')
+        sys.exit(3)
+
+    meminfo_huge_pages = meminfo['HugePages_Total']
+
+    if int(meminfo_huge_pages) < int(args.value):
+        exit_status = 2
+        output = (
+            f"/proc/meminfo['HugePages_Total'] == {meminfo_huge_pages} "
```
Comment (lamaral) [inline src/check_huge_pages.py:49]: Please prefix the output with `CRITICAL - `.

File: src/check_huge_pages.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            f"which is smaller than {args.value}"
+        )
+    else:
+        exit_status = 0
+        output = (
+            f"/proc/meminfo['HugePages_Total'] is {meminfo_huge_pages} "
```
Comment (lamaral) [inline src/check_huge_pages.py:55]: Please prefix the output with `OK - `.

File: src/check_huge_pages.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        print('Could not parse /proc/meminfo')
+        sys.exit(3)
+
+    meminfo_huge_pages = meminfo['HugePages_Total']
+
+    if int(meminfo_huge_pages) < int(args.value):
+        exit_status = 2
+        output = (
+            f"/proc/meminfo['HugePages_Total'] == {meminfo_huge_pages} "
+            f"which is smaller than {args.value}"
```
Comment (lamaral) [inline src/check_huge_pages.py:50]: Please make this line consistent with line 56 below.

File: src/check_huge_pages.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        )
+    else:
+        exit_status = 0
+        output = (
+            f"/proc/meminfo['HugePages_Total'] is {meminfo_huge_pages} "
+            f"which is >= {args.value}"
```
Comment (lamaral) [inline src/check_huge_pages.py:56]: Please make this line consistent with line 50 above.

File: src/check_huge_pages.py (hunk 1 above)
Comment (lamaral) [reply src/check_huge_pages.py:34]: I would also go as far as saying let's make this `--min_value`, so if we ever decide to check for max value, we don't have to change the script calls.

Comment (TheFRedFox) [inline src/check_huge_pages.py:34]:
```suggestion
                        help='The amount of huge pages to expect')
```

Comment (lamaral) [reply src/check_huge_pages.py:34]: Also a typo in `amount`

File: src/check_huge_pages.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+# THE SOFTWARE.
+
+import sys
+
+from argparse import ArgumentParser
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--value',
```
Comment (TheFRedFox) [inline src/check_huge_pages.py:33]: If there is no default, but based on the script it looks like a required argument, maybe it should be a positional parameter then? At least it should be configured as required.

**Review by lamaral — CHANGES_REQUESTED** [review]

File: src/check_huge_pages.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+def main(args):
+    meminfo = read_proc_meminfo()
+    if not meminfo:
+        print('Could not parse /proc/meminfo')
+        sys.exit(3)
+
+    meminfo_huge_pages = meminfo['HugePages_Total']
+
+    if int(meminfo_huge_pages) < int(args.value):
```
Comment (TheFRedFox) [inline src/check_huge_pages.py:46]: If `args.value` is of type `int` a cast is not needed anymore.

File: src/check_huge_pages.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    """Return /proc/meminfo as a dict"""
+    meminfo_dict = {}
+    try:
+        with open('/proc/meminfo', 'r') as proc_file:
+            for line in proc_file.readlines():
+                name, value = line.strip().replace(' kB', '').split(':', 1)
+                meminfo_dict[name] = int(value)
+            return meminfo_dict
+    except (OSError, IOError):
+        return None
```
Comment (TheFRedFox) [inline src/check_huge_pages.py:73]: Outside you are interpreting `None` return as parsing error. However, a `ValueError` should also then get caught here. `int(value)` and also the `name, value = line.[...].split(':', 1)` could raise a `ValueError` here.

File: src/check_huge_pages.py (hunk 6 above)
Comment (TheFRedFox) [inline src/check_huge_pages.py:33]: Also, shouldn't it be of type `int`?

File: src/check_huge_pages.py (hunk 2 above)
Comment (TheFRedFox) [inline src/check_huge_pages.py:49]: Prefixes like already Luiz said.

File: src/check_huge_pages.py (hunk 3 above)
Comment (meikomeik) [reply src/check_huge_pages.py:55]: prefixed

File: src/check_huge_pages.py (hunk 1 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Added your suggestions

File: src/check_huge_pages.py (hunk 2 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Added prefix

File: src/check_huge_pages.py (hunk 4 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Should be consistent now

File: src/check_huge_pages.py (hunk 5 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Should be consistent now

File: src/check_huge_pages.py (hunk 1 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: fixed typo

File: src/check_huge_pages.py (hunk 7 above)
Comment (meikomeik) [reply src/check_huge_pages.py:46]: Removed not needed int

File: src/check_huge_pages.py (hunk 6 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: changed it to int

File: src/check_huge_pages.py (hunk 2 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Done

File: src/check_huge_pages.py (hunk 8 above)
Comment (meikomeik) [reply src/check_huge_pages.py:71]: Added ValueError as part of the possible errors though each line always contains a colon folowed by a number so it should never throw a ValueError I think. Anyhow, why not also add ValueError

File: src/check_huge_pages.py (hunk 6 above)
Comment (meikomeik) [reply src/check_huge_pages.py]: Made the argument required

File: src/check_huge_pages.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    parser = ArgumentParser()
+    parser.add_argument('--min_value', type=int, required=True,
+                        help='The amount of huge pages to expect')
+    return parser.parse_args()
+
+
+def main(args):
+    meminfo = read_proc_meminfo()
+    if not meminfo:
+        print('Could not parse /proc/meminfo')
```
Comment (lamaral) [inline src/check_huge_pages.py:41]: I missed the `UNKNOWN - ` prefix here.

**Review by lamaral — APPROVED** [review]
There is just one more thing that I didn't see before. 
I'm pre-approving. Feel free to merge after fixing this last comment.

File: src/check_huge_pages.py (hunk 9 above)
Comment (meikomeik) [reply src/check_huge_pages.py:41]: fixed this one as well

**Pull request merged**